src/anomalib/pipelines/hpo/generator.py

```diff
@@ -0,0 +1,137 @@
+"""HPO sweep job generator."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from collections.abc import Generator
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from anomalib.data import get_datamodule
+from anomalib.models import get_model
+from anomalib.pipelines.components import JobGenerator
+from anomalib.pipelines.types import PREV_STAGE_RESULT
+from anomalib.utils.exceptions import try_import
+from anomalib.utils.logging import hide_output
+
+from .job import HPOJob
+from .utils import flatten_hpo_params, set_in_nested_dict
+
+if TYPE_CHECKING:
+    from lightning.pytorch.loggers import Logger
+
+logger = logging.getLogger(__name__)
+
+
+class HPOBackend(Enum):
+    """HPO backend."""
+
+    COMET = "comet"
+    WANDB = "wandb"
+
+    def __str__(self) -> str:
+        """Return the string representation."""
+        return self.value
+
+
+@dataclass
+class ParserConfig:
```

Comment:
Not sure if I fully understand this object. Is it parsing objects? or storing them in one dataclass?